### ISSUE TITLE
Strange sequence index produced in VKittiDataset

### ISSUE THREAD

**Issue by smileyenot983** [facebookresearch/vggt#353]
I am trying to run the training on KITTI dataset. It went nice for ~800 steps, but after that `seq_index` variable started taking some strange values:
                                                                        
INFO 2025-08-28 15:53:58,237 checkpoint.py:  65: Saving checkpoint at epoch 0 to logs/exp001/ckpts/checkpoint.pt                                                                                                                                            
Building dynamic dataloader with epoch: 0                                                                                                                                                                                                                   
seq_index: 5053                                                                                                                                                                                                                                             
seq_index: 53547                                                                                                                                                                                                                                            
seq_index: 35635                                                                                                                                                                                                                                            
seq_index: 10011                                                                                                                                                                                                                                            
seq_index: 46117                                                                                                                                                                                                                                            
seq_index: 71848                                                                                                                                                                                                                                            
seq_index: 8766                                                                                                                                                                                                                                             
seq_index: 65213                                                                                                                                                                                                                                            
seq_index: 34887seq_index: 52428seq_index: 41347                                                                                                                                                                                                            
seq_index: 45132                                                                                                                                                                                                                                            
seq_index: 68160                                                                                                                                                                                                                                            
                                                                                                                                                                                                                                                            
seq_index: 37497                                                                                                                                                                                                                                            
seq_index: 54747                                                                                                                                                                                                                                            
seq_index: 8195                                                                                                                                                                                                                                             
seq_index: 91255 


seq_index should be somewhere in [0,99] range. I am what is the issue

**Comment by jytime** [comment]
Did you get the number here?

https://github.com/facebookresearch/vggt/blob/059c288227aa00faaf1a4ecf3f4f3be9b40aa467/training/data/datasets/vkitti.py#L110-L116

**Comment by OumaimaCharia** [comment]
Hey, I am running into the same error: index out of range. Actually, the index is within the range (0–9). 
I am getting the number from [vggt/training/data/datasets/vkitti.py](https://github.com/facebookresearch/vggt/blob/059c288227aa00faaf1a4ecf3f4f3be9b40aa467/training/data/datasets/vkitti.py#L110-L116)

**Comment by smileyenot983** [comment]
it's because the training is False in eval mode and thus the range is not fixed to sequence length

**Comment by smileyenot983** [comment]
> Did you get the number here?
> 
> [vggt/training/data/datasets/vkitti.py](https://github.com/facebookresearch/vggt/blob/059c288227aa00faaf1a4ecf3f4f3be9b40aa467/training/data/datasets/vkitti.py#L110-L116)
> 
> Lines 110 to 116 in [059c288](/facebookresearch/vggt/commit/059c288227aa00faaf1a4ecf3f4f3be9b40aa467)
> 
>  if self.inside_random and self.training: 
>      seq_index = random.randint(0, self.sequence_list_len - 1) 
>   
>  if seq_name is None: 
>      seq_name = self.sequence_list[seq_index] 
>   
>  camera_id = int(seq_name[-1])

yes
